models/run.py

```diff
@@ -48,96 +140,130 @@
+                # go about your day accruing encounters and clustering them
+                for idx, encounter in enumerate(encounters):
+                    encounter_time = encounter['time']
+                    unobs = encounter['payload']['unobserved']
+                    encountered_human = hd[unobs['human2']['human_id']]
+                    message = encountered_human.cur_message(current_day, RiskModel)
+                    encountered_human.sent_messages[
+                        str(unobs['human1']['human_id']) + "_" + str(encounter_time)] = message
+                    human.messages.append(message)
+                    got_exposed = encounter['payload']['unobserved']['human1']['got_exposed']
+                    if got_exposed:
+                        human.exposure_message = _encode_message(message)
+
+            daily_output = parallel((delayed(proc_human)(params) for params in all_params))
+
+            # if the encounter happened within the last 14 days, and your symptoms started at most 3 days after your contact
+            if RiskModel.quantize_risk(human.start_risk) != RiskModel.quantize_risk(human.risk):
+                for k, m in human.sent_messages.items():
+                    if current_day - m.day < 14:
+                        hd[m.unobs_id].update_messages.append(human.cur_message_risk_update(m.day, m.risk, RiskModel))
+
+            for idx, output in enumerate(daily_output):
+                hd[output['human'].name] = output['human']
+                del daily_output[idx]['human']
+        all_outputs.append(daily_output)
+
+        print(f"mainloop {time.time() - start1}")
+
+        # add risks for plotting
+        todays_date = start + datetime.timedelta(days=current_day)
+        daily_risks.extend([(np.e ** human.risk, human.is_infectious(todays_date)[0], human.name) for human in hd.values()])
+        if args.plot_daily:
+            hist_plot(daily_risks, f"{args.plot_path}day_{str(current_day).zfill(3)}.png")
+        all_risks.extend(daily_risks)
+
+    if args.save_training_data:
+        pickle.dump(all_outputs, open(args.output_file, 'wb'))
+
+    dist_plot(all_risks,  f"{args.plot_path}all_risks.png")
 
     # make a gif of the dist output
-    process = subprocess.Popen(f"convert -delay 50 -loop 0 {PATH_TO_PLOT}/*.png {PATH_TO_PLOT}/risk.gif".split(), stdout=subprocess.PIPE)
+    process = subprocess.Popen(f"convert -delay 50 -loop 0 {args.plot_path}/*.png {args.plot_path}/risk.gif".split(), stdout=subprocess.PIPE)
     output, error = process.communicate()
 
     # write out the clusters to be processed by privacy_plots
-    if hd[0].M:
-        clusters = []
-        for human in hd.values():
-            clusters.append(human.M)
-        json.dump(clusters, open(CLUSTER_PATH, 'w'))
+    clusters = []
+    for human in hd.values():
+        clusters.append(human.M)
+    json.dump(clusters, open(args.cluster_path, 'w'))
 
+
+if __name__ == "__main__":
+    args = parser.parse_args()
```

Comment:
This can be moved inside main()

Reply:
made parse_args its own function -- is this what you meant? I thought it was cleaner to have it seperate (and therefore independently testable)

Reply:
done